core/01_core.js

```diff
@@ -455,6 +455,65 @@
     op_is_weak_set,
   } = ensureFastOps();
 
+  function propWritable(value) {
+    return {
+      value,
+      writable: true,
+      enumerable: true,
+      configurable: true,
+    };
+  }
+
+  function propNonEnumerable(value) {
+    return {
+      value,
+      writable: true,
+      enumerable: false,
+      configurable: true,
+    };
+  }
+
+  function propReadOnly(value) {
+    return {
+      value,
+      enumerable: true,
+      writable: false,
+      configurable: true,
+    };
+  }
+
+  function propGetterOnly(getter) {
+    return {
+      get: getter,
+      set() {},
+      enumerable: true,
+      configurable: true,
+    };
+  }
+
+  function propNonEnumerableLazyLoaded(getter, loadFn) {
+    let valueIsSet = false;
+    let value;
+
+    return {
+      get() {
+        loadFn();
+        if (valueIsSet) {
+          return value;
+        } else {
+          return getter();
+        }
+      },
+      set(v) {
+        loadFn();
+        valueIsSet = true;
+        value = v;
+      },
+      enumerable: false,
+      configurable: true,
+    };
+  }
+
```

Comment:
can also https://github.com/denoland/deno/blob/main/runtime/js/98_global_scope_shared.js#L65-L88 be added? propWritable lazy load counterpart